spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/ModuleAggregationUtils.java

```diff
@@ -0,0 +1,154 @@
+		LinkedHashMap<String, ConfigurableApplicationContext> children = createChildContexts(parentContext, modules, args);
+		for (Map.Entry<String, ConfigurableApplicationContext> childContextEntry : children.entrySet()) {
+			ConfigurableApplicationContext childContext = childContextEntry.getValue();
+			parentContext.getBeanFactory().registerSingleton(childContextEntry.getKey(),
+					new BindableContextWrapper(childContext));
+			childContext.setParent(parentContext);
+		}
+	}
+
+	private static ConfigurableApplicationContext createParentContext(String[] args) {
+		SpringApplicationBuilder aggregatorParentConfiguration
+				= new SpringApplicationBuilder();
+		aggregatorParentConfiguration.sources(AggregatorParentConfiguration.class)
+				.web(false)
+				.headless(true)
+				.properties("spring.jmx.default-domain=" + UUID.randomUUID().toString().replace("-", ""))
+				.listeners(new UnbindOnCloseApplicationListener());
+		return aggregatorParentConfiguration.run(args);
+	}
+
+	private static LinkedHashMap<String, ConfigurableApplicationContext> createChildContexts(ConfigurableApplicationContext applicationContext, Class<?>[] modules,
+																																										 String args[][]) {
+		LinkedHashMap<String, ConfigurableApplicationContext> children = new LinkedHashMap<>();
+		for (int i = modules.length - 1; i >= 0; i--) {
+			String moduleClassName = modules[i].getName();
+			SpringApplicationBuilder childModuleBuilder = embedModule(applicationContext, getNamespace(moduleClassName, i), modules[i]);
+			ConfigurableApplicationContext childContext = childModuleBuilder.run(args != null ? args[i] : new String[0]);
+			children.put(getNamespace(moduleClassName,i), childContext);
+		}
+		return children;
+	}
+
+	private static String getNamespace(String moduleClassName, int index) {
+		return moduleClassName + "_" + index;
+	}
+
+	private static SpringApplicationBuilder embedModule(ConfigurableApplicationContext applicationContext, String namespace,
+																										 Class<?> module) {
+		return new SpringApplicationBuilder(module)
+				.web(false)
+				.showBanner(false)
+				.properties(BindableProxyFactory.CHANNEL_NAMESPACE_PROPERTY_NAME + "=" + namespace)
+				.registerShutdownHook(false)
+				.parent(applicationContext);
+	}
+
+	private static void prepareSharedChannelRegistry(SharedChannelRegistry sharedChannelRegistry, Class<?>[] modules) {
+		DirectChannel sharedChannel = null;
+		for (int i = 0; i < modules.length; i++) {
+			Class<?> module = modules[i];
+			String moduleClassName = module.getName();
+			if (i > 0) {
+				sharedChannelRegistry.putSharedChannel(getNamespace(moduleClassName,i) + ".input", sharedChannel);
+			}
+			sharedChannel = new DirectChannel();
+			if (i < modules.length - 1) {
+				sharedChannelRegistry.putSharedChannel(getNamespace(moduleClassName,i) + ".output", sharedChannel);
+			}
+		}
+	}
+
+	/**
+	 * Basic configuration for a parent
+	 */
+	@EnableAutoConfiguration
+	@EnableModule
```

Comment:
Actually I was convinced on having `@EnableModule` :-) My intention in this comment was to avoid calling the lifecycle beans prematurely to avoid any surprises in future. I was thinking to have some kind of exclude (based on some conditions) specifically on this lifecycle bean.